Alerting: Make QueryEditor not collapsable

VikaCep opened the pull request:
**What is this feature?**

Makes the `QueryEditor` not collapsable when creating/editing an alert rule. 

**Who is this feature for?**

All users

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/68191

**Special notes for your reviewer:**

![image](https://github.com/grafana/grafana/assets/6271380/a20edcd2-bada-40e6-9055-3f6151c52bb1)

VikaCep committed:
Add option to make QueryEditor not collapsable

VikaCep committed:
Make QueryEditor not collapsable in Alerting

torkelo reviewed (commented):
While you are at it can you change the border around QueryWrapper from medium to weak? the medium border is really only for inputs, everything else should use weak border

VikaCep committed:
Change query editor border to weak

The medium border is really only for inputs, everything else should use weak border

VikaCep committed:
Add tests

soniaAguilarPeiron reviewed (approved):
LGTM!
